Source/Core/Core/GeckoCodeConfig.cpp

```diff
@@ -251,4 +250,21 @@ void SaveCodes(IniFile& inifile, const std::vector<GeckoCode>& gcodes)
   inifile.SetLines("Gecko_Enabled", enabled_lines);
   inifile.SetLines("Gecko_Disabled", disabled_lines);
 }
+
+bool DeserializeLine(const std::string& line, GeckoCode::Code* out)
+{
+  std::vector<std::string> pieces = SplitString(line, ' ');
+  bool good = false;
+
+  if (pieces.size() > 0)
+    good &= TryParse(pieces[0], &out->address, 16);
+
+  if (pieces.size() > 1)
+    good &= TryParse(pieces[1], &out->data, 16);
+
+  out->original_line = line;
```

Comment:
Should we trust the caller that `out` was initialized and points to something? Just to be on the safe side.

Reply:
Is there anything reasonable to do if it doesn't point to valid memory?

Reply:
Fair enough. I mean, you should probably `return false` when `out == nullptr`. Not sure if you could even detect an invalid pointer. Probably should've worded it clearer in first place (wrt `out` being `null`.

Reply:
I don't think it's worth checking for, since there's no case where passing in `nullptr` would be useful for the caller.

Reply:
_Intentionally_, yes. But your choice, if you're certain it usually ends up being fine.